src/validator.js

```diff
@@ -29,47 +41,83 @@ export default class Validator {
   }
 
   validateField(field, value, values = {}) {
-    let result = null;
+    let error = null;
     if (this.config.hasOwnProperty(field)) {
-      const fieldRules = keys(this.config[field]);
-      for (let i = 0; i < fieldRules.length; i++) {
-        const rule = fieldRules[i];
-        const options = { values };
-        if (isBoolean(this.config[field][rule])) {
-          options.if = constant(this.config[field][rule]);
-        }
+      error = this.validateRule(
+        'required',
+        field,
+        value,
+        { values, required: true }
+      );
+      const empty = !!error;
+
+      if (error !== null && !this.evaluateIf(field, 'required', values)) {
+        error = null;
+      }
 
-        result = this.validateRule(
-          rule,
-          field,
-          value,
-          assign({}, this.config[field][rule], options)
-        );
-
-        if (result) {
-          result = { field, rule: result, value, config: this.config[field] };
-          this.errors = assign({}, this.errors, { [field]: result });
-          break;
-        } else {
-          if (this.errors.hasOwnProperty(field)) {
-            this.errors = omit(this.errors, field);
+      if (!empty) {
```

Comment:
> First of all, there is a lot of negation to get to the value of this. Can we flip the semantics of this variable? That would remove two negations.

Yes 👍 

>Second: Is this the case when the validateRule function reports that the rule passes validation? If so could we rename the variable to somthing along the lines of isValid?

Not necessarily, if it is required then it fails validation if it is not required it passes validation. The if statement is just to make sure we do not run any other rules if required fails.